source/FallingBlock.hx

```diff
@@ -0,0 +1,80 @@
+package;
+
+import flixel.FlxObject;
+import flixel.FlxSprite;
+import flixel.system.FlxAssets.FlxGraphicAsset;
+import flixel.util.FlxColor;
+import haxe.Timer;
+
+/**
+ * A block that falls after being stepped on
+ * @author TR Nale
+ */
+class FallingBlock extends Block
+{
+	static var FALLACCEL:Float = 100;
+	static var MAXSPEED:Float = 500;
+	static var DESTRUCTIONLINE = 500;
+	public var fallDelay:Int;
+	private var isFalling:Bool;
+	
+	/**
+	 * Constructor
+	 * 
+	 * @param	X	X coordinate in tilemap
+	 * @param	Y	Y coordinate in tilemap
+	 * @param	SimpleGraphic	Non-animating graphic.
+	 * @param	Breakable	Whether the block breaks when hit from below; defaults to false
+	 */
+	public function new(?X:Float=0, ?Y:Float=0, ?SimpleGraphic:FlxGraphicAsset, ?Delay:Int=1000) 
+	{
+		super(X, Y, SimpleGraphic);
+		fallDelay = Delay;
+		isFalling = false;
+		maxVelocity.y = MAXSPEED;
+	}
+	
+	/**
+	 * Destructor
+	 */
+	override public function destroy()
+	{
+		fallDelay = null;
+		isFalling = null;
+		super.destroy();
+	}
+	
+	/**
+	 * Causes block to fall after a delay when touched from the top
+	 */
+	override public function onTouch()
+	{
+		if (isTouching(FlxObject.UP))
+		{
+			color = FlxColor.YELLOW;
```

Comment:
When I built and ran the game on a school computer, this recolor wasn't working properly. Instead of changing the block to yellow, it turns black, which makes it very hard to see against the background.